### PR TITLE
Added combat picking

### PR DESCRIPTION
**Pull request by dr-leustyin**

### CODE REVIEW

**Comment by Sarvatt** [comment]
Could you add Locks to https://github.com/rpherbig/dr-scripts/blob/master/validate.lic#L381 also?

File: combat-trainer.lic [hunk 1]
```diff
@@ -1779,6 +1782,32 @@ class TrainerProcess
       bput(%w[weave bob circle].sample, 'roundtime', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') unless game_state.npcs.empty?
     when 'Analyze'
       analyze(game_state, 0)
+    when 'Locks'
+      return if game_state.loaded
```
Comment (rpherbig) [inline combat-trainer.lic]: 💡 I highly recommend extracting this into a helper method (leaving this function to mostly be managing control flow)

File: combat-trainer.lic [hunk 2]
```diff
@@ -1779,6 +1782,32 @@ class TrainerProcess
...
+      return if left_hand
+      boxes = get_boxes(@pet_box_source)
+      return if boxes.length == 0
+      box = boxes.first
+      # Check what's in the right hand so we can get it back
+      rhitem = @equipment_manager.item_by_desc(right_hand) if right_hand
```
Comment (rpherbig) [inline combat-trainer.lic]: ❓ Wouldn't the `rhitem` normally be a weapon? We have a method on `game_state` that can 'get rid' of the weapon safely (and one that gets it back).

File: combat-trainer.lic [hunk 3]
```diff
@@ -1779,6 +1782,32 @@ class TrainerProcess
...
+      @equipment_manager.stow_weapon("#{rhitem.adjective} #{rhitem.name}") if rhitem
+      return if right_hand  # Couldn't empty right hand for some reason (typically a summoned weapon).  No picking allowed
```
Comment (rpherbig) [inline combat-trainer.lic]: ❓ If you check the summoning code, there's some logic to check if your current weapon is summoned. I suggest checking for that, outputting a message, and returning. What do you think?

File: combat-trainer.lic [hunk 4]
```diff
@@ -1779,6 +1782,32 @@ class TrainerProcess
...
+      bput("get #{box} from my #{@pet_box_source}","You get", "What were")
+      retreat
+      response =  bput("pick my #{box} blind", 'not even locked', 'Roundtime', 'Find a more appropriate tool','You realize that would be next to impossible')
+      if response == 'not even locked'
+        # Dump the box so that combat-trainer lootprocess can take over
+        fput("open my #{box}")
```
Comment (rpherbig) [inline combat-trainer.lic]: ❓ Could any or all of these three be converted to the more reliable `bput`?

Comment (rpherbig) [reply combat-trainer.lic]: And/or line 1810

File: combat-trainer.lic [hunk 5]
```diff
@@ -1779,6 +1782,32 @@ class TrainerProcess
       bput(%w[weave bob circle].sample, 'roundtime', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') unless game_state.npcs.empty?
     when 'Analyze'
       analyze(game_state, 0)
+    when 'Locks'
+      return if game_state.loaded
+      return if left_hand
+      boxes = get_boxes(@pet_box_source)
```
Comment (rpherbig) [inline combat-trainer.lic]: ❓ Would you want to check if `@pet_box_source` is defined?

File: validate.lic [hunk 6]
```diff
@@ -378,7 +378,7 @@ class DRYamlValidator
     return unless settings.training_abilities
 
     settings.training_abilities
-            .reject { |skill, _cooldown| ['PercMana', 'Perc', 'Perc Health', 'Astro', 'App', 'App Quick', 'App Careful', 'Tactics', 'Hunt', 'Pray', 'Scream', 'Khri Prowess', 'Stealth', 'Ambush Stun', 'Ambush Choke', 'Favor Orb', 'Charged Maneuver', 'Meraud', 'Recall', 'Barb Research Debilitation', 'Barb Research Augmentation', 'Barb Research Warding', 'Collect', 'Analyze', 'PrayerMat', 'Smite'].include?(skill) }
+            .reject { |skill, _cooldown| ['PercMana', 'Perc', 'Perc Health', 'Astro', 'App', 'App Quick', 'App Careful', 'Tactics', 'Hunt', 'Pray', 'Scream', 'Khri Prowess', 'Stealth', 'Ambush Stun', 'Ambush Choke', 'Favor Orb', 'Charged Maneuver', 'Meraud', 'Recall', 'Barb Research Debilitation', 'Barb Research Augmentation', 'Barb Research Warding', 'Collect', 'Analyze', 'PrayerMat', 'Smite', 'Locks'].include?(skill) }
```
Comment (rpherbig) [inline validate.lic]: ❓ Similar to the above question, do you want to add a new test that if `Locks` is in `training_abilities`, `pet_box_source` is also defined?

Comment (Sarvatt) [reply validate.lic]: I can follow up adding that after if this is a blocker.

Comment (rpherbig) [reply validate.lic]: 👍

File: combat-trainer.lic (hunk 3 above)
Comment (dr-leustyin) [reply combat-trainer.lic]: Holding a summoned weapon is one of the things that broke the logic when I was first working on / testing this.  I'm not sure if there are other things that might also get in the way that might not be summoned weapons.

I think breaking the pick execution if there's something unexpected in your hands is a safe guard

Comment (rpherbig) [reply combat-trainer.lic]: :+1:

**Pull request merged**